src/hotspot/share/gc/shared/gcLocker.cpp

```diff
@@ -112,10 +121,16 @@ void GCLocker::stall_until_clear() {
     log_debug_jni("Allocation failed. Thread stalled by JNI critical section.");
   }
 
+  elapsedTimer elapsed_timer;
```

Comment:
In GC code we tend to use the newer `Ticks` and `Tickspan` API, not `elapsedTimer`. Only Parallel GC uses it at this point afaict (or just `os::elapsedTime()`/`os::elapsed_counter()`).

Maybe it's even worth to add a special class that can be used with scopes to hide all that including the manual call to `log_debug_jni` (automatically done in the destructor). Probably not really useful.

Reply:
Addressed in latest revision.